engine/collection/message_hub/message_hub.go

```diff
@@ -382,6 +379,8 @@ func (h *MessageHub) OnOwnVote(blockID flow.Identifier, view uint64, sigData []b
 	}
 	if ok := h.ownOutboundVotes.Push(packed); ok {
 		h.ownOutboundMessageNotifier.Notify()
+	} else {
+		h.engineMetrics.MessageDropped(metrics.EngineCollectionMessageHub, metrics.MessageClusterBlockVote)
```

Comment:
This is changing convention slightly from previous usage of these metric labels, and other usages of `MessageDropped`. Usually it means, we received a message from another node, but dropped it before we processed it. Here it means, we are preparing to transmit a message we created, but dropped it before we could transmit it. I think it would be beneficial to use a label name that made this clear, and differentiated this case. 
```suggestion
		h.engineMetrics.MessageDropped(metrics.EngineCollectionMessageHub, metrics.MessageOutboundClusterBlockVote)
```

Reply:
I was thinking about it a bit differently. Cluster's `MessageHub` drops block vote so the respective metric message will be:
`metrics.EngineCollectionMessageHub` + `metrics.MessageClusterBlockVote` since it's the only vote that message hub is dropping it means it has to be an outbound block vote. I mean we can use the metrics more compactly this way instead of spawning a label for each use case which is harder to support in future not even talking about on-boarding someone into to the topic.

Reply:
> since it's the only vote that message hub is dropping it means it has to be an outbound block vote

It's true that it has to be an outbound vote, but it isn't at all obvious without familiarity with the implementation details. At a high level, `MessageHub` both accepts and transmits votes - it would be completely reasonable for someone with high-level familiarity with the compliance engine design, and experience with message received/processed/dropped metrics reporting inbound messages, to assume this is referring to an inbound vote. Or at least for it to be unclear without further investigation. 

Another perspective is we have both message received and sent metrics so we can differentiate those cases - we could do the same for dropped messages: `InboundMessageDropped`, `OutboundMessageDropped`.

Reply:
Agree with Jordan here, that the differentiation between inbound and outbound messages is too subtle. 

I can see two pragmatic ways how we can add sufficient differentiation:
1. Jordan's proposal
   > we have both message received and sent metrics so we can differentiate those cases - we could do the same for dropped messages: `InboundMessageDropped`, `OutboundMessageDropped`.

    i.e. we would differentiate by having _different metrics_
2. Alternatively, we could keep the same metric, but differentiate on the message type. Specifically, we could use different `message` strings here: https://github.com/onflow/flow-go/blob/628b20cd267f34864960ab5834fdaeabddab7900/module/metrics.go#L91-L93

    For collector nodes, we would split these message types https://github.com/onflow/flow-go/blob/628b20cd267f34864960ab5834fdaeabddab7900/module/metrics/labels.go#L120-L122
      and introduce one dedicated type for inbound and another for outbound, e.g

    ```golang
    MessageInboundClusterBlockProposal = "inbound_cluster_proposal"
    MessageOutboundClusterBlockProposal = "outbound_cluster_proposal"
    ```

Reply:
I think I will stick to the approach 1) I would like to reuse as much as possible existing labels and avoid introducing more of them to reduce maintenance complexity. Additionally I would like to remove cluster labels(`MessageClusterConsensusObject`) and use instead `MessageConsensusObject` because the differentiation between consensus and collector engines comes from _engine_ label.

Reply:
Separate metrics for outbound and inbound messages: https://github.com/onflow/flow-go/pull/3623/commits/a2902ae90b8a1e39058f3708b8052b8e4af2dfab
Cleanup of cluster specific messages: https://github.com/onflow/flow-go/blob/628b20cd267f34864960ab5834fdaeabddab7900/engine/consensus/compliance/core.go#L319